reps/2023-03-20-air-storage-path.md

```diff
@@ -0,0 +1,155 @@
+## Summary
+
+We'd like to change how users of Ray AIR (specifically: Train + Tune) configure permanent storage locations.
+
+Current way:
+- Users can specify `RunConfig.local_dir` or `tune.run(local_dir)` for storage on the local node
+- Users can specify `SyncConfig.upload_dir` for storage on remote nodes
+
+Future way:
+- We have a unified setting `RunConfig.storage_path` that can be set either to a local dir or a remote path.
+- Setting the storage path to a cloud or NFS URI (e.g., `s3://`, or `file://` that points to a NFS mount). In these cases, data will be first written to a local cache dir on the worker, and then synced to a subdirectory in the storage path designated by `<experiment_name>/<trial_name>/`.
+- Setting the storage path to a purely local URI (e.g., `/home/foo/ray_results`). In this mode, trial data is synced to the head node via the object store. We would generally recommend using a remote storage path or shared directory instead.
+
+
+```python
+import ray
+from ray import air, tune
+
+results = tune.Tuner(train_fn, run_config=air.RunConfig(storage_path="s3://foo/bar")).fit()
+assert results.best_checkpoint().path.startswith("s3://foo/bar")
+```
+
+### General Motivation
+
+Users care about where their data is ultimately stored. They usually don't care about
+how it got there.
+
+However, currently they specify a local storage directory, from which data is periodically "synced"
+to a remote storage location. This flow (collect locally, sync to storage) is more of an implementation
+detail.
+
+Instead, users should just specify where their data ends up - with `storage_path`.
```

Comment:
this sounds good. a quick question is what happens to the metadata and metrics that end up on the driver node today?
if users only specify an s3 path, where would they go under the new scheme.
also I assume the driver process can also upload things to specified s3 paths right? because there will be Trial level artifacts that folks need to save.

Reply:
ok, cool!